src/utils/resolve-config.ts

```diff
@@ -47,19 +50,25 @@ export function resolveConfig(inlineConfig: Partial<Config>): Config {
   return { ...mergedConfig, scopes: getScopes() };
 }
 
-function flatFileConfig({keysManager, rootTranslationsPath, langs, scopePathMap}: TranslocoGlobalConfig): Partial<Config> {
-    if (keysManager?.input) {
-      keysManager.input = Array.isArray(keysManager.input)
-          ? keysManager.input
-          : keysManager.input.split(',')
+function flatFileConfig({
+  keysManager,
+  rootTranslationsPath,
+  langs,
+  scopePathMap,
+}: TranslocoGlobalConfig): Partial<Config> {
+  if (keysManager?.input) {
+    keysManager.input = Array.isArray(keysManager.input)
+      ? keysManager.input
+      : keysManager.input.split(',');
   }
 
   return {
-    translationsPath: rootTranslationsPath,
-    langs,
-    scopePathMap,
-    ...(keysManager as Omit<TranslocoGlobalConfig['keysManager'], 'input'> & Pick<Config, 'input'>),
-  }
+    ...(rootTranslationsPath ? { translationsPath: rootTranslationsPath } : {}),
+    ...(langs ? { langs } : {}),
+    ...(scopePathMap ? { scopePathMap } : {}),
```

Comment:
Can you explain?

Reply:
Exactly the same reasoning as [here](https://github.com/jsverse/transloco-keys-manager/pull/177#discussion_r1494818512)

Reply:
[commented](https://github.com/jsverse/transloco-keys-manager/pull/177#discussion_r1495455639)